src/archivist/recorder/index.js

```diff
@@ -0,0 +1,86 @@
+export default class Recorder {
+  constructor({ versionsStorageAdapter, snapshotsStorageAdapter }) {
+    if (!versionsStorageAdapter && !snapshotsStorageAdapter) {
+      throw new RangeError('storage adapters should be defined both for versions and snapshots');
+    }
+
+    this.versionsStorageAdapter = versionsStorageAdapter;
+    this.snapshotsStorageAdapter = snapshotsStorageAdapter;
+  }
+
+  async initialize() {
+    return Promise.all([ this.versionsStorageAdapter, this.snapshotsStorageAdapter ].map(storageAdapter => {
+      if (typeof storageAdapter.initialize != 'function') {
+        return Promise.resolve();
+      }
```

Comment:
I am not sure this is useful: we expect an adapter to comply with an API that includes `initialize`, that's all. If we want to be smart, it should be by providing an abstract Storage Adapter class that implements a no-op for both `initialize` and `finalize`, not in clients.